### ISSUE TITLE
Remove spell check from git "New Branch"

### ISSUE THREAD

**Issue by njtierney** [rstudio/rstudio#2027]
### System details

    OS Version      : Max OS darwin15.6.0
    RStudio Version : 1.1.383
    R Version       : 3.4.3

### Steps to reproduce the problem

Go to "Git" tab and click "New Branch", enter new branch name

### Describe the problem in detail

When I enter a branch name, an autocorrect suggestion appears, which interferes with creating a branch (see below)

<img width="391" alt="image" src="https://user-images.githubusercontent.com/6488485/35203112-a264d678-ff7a-11e7-848c-d983e81faedf.png">

### Describe the behavior you expected

I expect that when providing a new branch name, there is no spellcheck, as this is a place where spelling is likely to be deliberate and also not a regular english word.

```
> sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.2

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] naniar_0.1.5.9500 testthat_2.0.0    devtools_1.13.4  

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15       rstudioapi_0.7     bindr_0.1          knitr_1.18         magrittr_1.5      
 [6] munsell_0.4.3      colorspace_1.3-2   R6_2.2.2           rlang_0.1.6        plyr_1.8.4        
[11] dplyr_0.7.4        tools_3.4.3        grid_3.4.3         gtable_0.2.0       withr_2.1.1       
[16] yaml_2.1.16        lazyeval_0.2.1     digest_0.6.14      assertthat_0.2.0   tibble_1.4.1      
[21] bindrcpp_0.2       purrr_0.2.4        ggplot2_2.2.1.9000 glue_1.2.0         memoise_1.1.0     
[26] compiler_3.4.3     pillar_1.1.0       scales_0.5.0.9000  visdat_0.2.2       pkgconfig_2.0.1   
```
Thanks again for making something I use every day! This is pretty minor but thought it might be worthwhile noting.

**Comment by ronblum** [comment]
This appears to be limited to RStudio MacOS Desktop.

Reproduced on:
RStudio Desktop 1.1.414 on MacOS 10.12.6.

Not an issue on:
Chrome on Mac 10.12.5 pointing at RStudio Server 1.1.414 on Ubuntu 16.04
RStudio Desktop 1.1.414 on Windows 10

**Comment by kevinushey** [comment]
Thanks for the clear bug report!

**Comment by njtierney** [comment]
Thanks for providing a great template and awesome system, see you at the rstudioconf!

**Comment by dfalty** [comment]
Verified fixed 1.2.359.
